Algorithms-Binary-Search.md

```diff
@@ -0,0 +1,82 @@
+# Binary search
+
+Binary search is a search algorithm that finds the position of a target value within a sorted array.
+
+## Example
+
+![Binary search](https://upload.wikimedia.org/wikipedia/commons/f/f7/Binary_search_into_array.png)
+
+The above illustration shows the working of the binary search algorithm on a sorted array when the target value is **4**.
+
+## Algorithm
+
+Binary search works on sorted arrays. A binary search begins by comparing the middle element of the array with the target value. If the target value matches the middle element, its position in the array is returned. If the target value is less or more than the middle element, the search continues the lower or upper half of the array respectively with a new middle element, eliminating the other half from consideration.
```

Comment:
@atjonathan: Quoting, "Its, without an apostrophe, is the possessive of the pronoun it. It’s, with an apostrophe, is a contraction of it is or it has. If you’re not sure which spelling to use, try replacing it with it is or it has. If neither of those phrases works in its place, then its is the word you’re looking for.".

- [Read more](http://grammarist.com/spelling/its-its/)

Sorry for the "Grammar Nazi" behavior 😄